test/new-e2e/tests/containers/k8s_test.go

```diff
@@ -209,7 +195,7 @@ func (suite *k8sSuite) Test00UpAndRunning() {
 			if suite.NoError(err) && len(linuxPods.Items) >= 1 {
 				stdout, stderr, err := suite.podExec("datadog", linuxPods.Items[0].Name, tt.container, []string{"agent", "version"})
 				if suite.NoError(err) {
```

Comment:
What happens if err is not nil?

Reply:
If `err` is not `nil`, `suite.NoError(err)` will “mark the test as failed”. The `err` error will also be logged with a bunch of other information like the source file and line number for ex. in the test suite report.

As `suite.NoError(err)` invokes `assert.NoError(suite.T(), err)` behind the hood and not `require.NoError(suite.T(), err)`, the function isn’t immediately stopped. Instead, although the verdict is already known (it will be a fail), the function keeps on being executed.
In order to make the test function abort immediately, we could have used `suite.Require().NoError(err)` instead.

But because the test failed, `suite.NoError(err)` returns false and the `if` block isn’t executed.
If this precise case, if the `kubectl pod exec` failed, we can skip the piece of code that checks the output.